x/mongo/driver/connstring/connstring.go

```diff
@@ -254,14 +255,25 @@ func (p *parser) parse(original string) error {
 		hosts = uri[:idx]
 	}
 
-	var connectionArgsFromTXT []string
 	parsedHosts := strings.Split(hosts, ",")
+	uri = uri[len(hosts):]
```

Comment:
The Git diff of this file might be hard to read. I had to reorganize the connection argument parsing logic slightly. We now

1. Extract the database from the URI
2. Grab connection arguments from the URI
3. If SRV scheme is being used:
    1. Grab connection arguments from TXT record
    2. Enable SSL by default
4. Apply connection arguments
5. If SRV scheme is being used:
    1. Do SRV lookup

I needed access to the custom SRV service name (`srvServiceName`) for SRV lookup, so I had to move all connection argument processing before the lookup.